string class_name
int64 id
int64 x1
int64 y1
int64 x2
int64 y2
int64 cx
int64 cy